metagraph/src/graph/annotated_graph_algorithm.hpp

```diff
@@ -4,66 +4,57 @@
 #include <vector>
 #include <functional>
 
-#include "common/threads/threading.hpp"
-#include "common/vectors/bitmap.hpp"
 #include "graph/annotated_dbg.hpp"
-#include "graph/representation/masked_graph.hpp"
+#include "common/vectors/bitmap.hpp"
 
 
 namespace mtg {
 namespace graph {
 
-typedef std::function<size_t()> LabelCountCallback;
 
-template <typename NodeType>
-class Alignment;
+class MaskedDeBruijnGraph;
+
+// A container for differential assembly parameters. These are:
+// label_mask_in_kmer_fraction: minimum fraction of in-labels which should be
+//                              present for a node to be kept
+// label_mask_in_unitig_fraction: minimum fraction of nodes within a unitig which
+//                                must satisfy the in-label criteria
+// label_mask_out_kmer_fraction: maximum fraction of out-labels which can be
+//                               present in a kept node
+// label_mask_out_unitig_fraction: maximum fraction of nodes within a unitig which
+//                                 can satisfy the out-label criteria
+// label_mask_other_unitig_fraction: maximum fraction of nodes within a unitig which
+//                                   can contain labels not in the in- and out-groups
+// add_complement: also consider the reverse complements of nodes
+struct DifferentialAssemblyConfig {
+    double label_mask_in_unitig_fraction = 1.0;
+    double label_mask_in_kmer_fraction = 1.0;
+    double label_mask_out_unitig_fraction = 0.0;
+    double label_mask_out_kmer_fraction = 0.0;
+    double label_mask_other_unitig_fraction = 1.0;
+    bool add_complement = false;
+};
+
+std::pair<sdsl::int_vector<>, std::unique_ptr<bitmap>>
+fill_count_vector(const AnnotatedDBG &anno_graph,
+                  const std::vector<AnnotatedDBG::Annotator::Label> &labels_in,
+                  const std::vector<AnnotatedDBG::Annotator::Label> &labels_out,
+                  size_t num_threads,
+                  bool update_in_place,
+                  const sdsl::int_vector<> *init_counts = nullptr);
```

Comment:
I removed `update_in_place` from the interface since it can be computed internally. Basically, it's set to true if the underlying graph is DBGSuccinct because the implementation of `call_sequences` for masked DBGSuccinct makes a copy of a bitmask, so `update_in_place` is used to prevent making another bitmask.